### PR TITLE
Make it easier to create a manual snsdemo release

### PR DESCRIPTION
**Pull request by dskloetd**
# Motivation

snsdemo releases are automatically created once a week.
In order to create a release outside the schedule, one has to perform some manual steps.
Since we already have a workflow that can create a release let's also use that for manual release.

# Changes

1. Add a checkbox when starting the workflow manually to indicate a release should be created.
2. Create a release when the checkbox is checked in addition to when the workflow runs on schedule.
3. Remove the functionality to create a release from branch `snapshot-schedule`. This was a branch created by @bitdivine supposedly for testing.
4. If a release was already created on the same day, try appending `.1` or `.2` etc., similar to what ic-js and gix-components do.

# Tested

Checkbox appears:
image

Created this release from this branch: https://github.com/dfinity/snsdemo/releases/tag/release-2024-05-10

### CODE REVIEW

File: .github/workflows/snapshot.yml [hunk 1]
```diff
@@ -42,20 +47,22 @@ jobs:
...
-            git tag "${tag_name}"
-            git push origin "refs/tags/${tag_name}"
-          fi
+          index=0
+          while git show-ref --tags --verify --quiet "refs/tags/${tag_name}"; do
+            echo "Tag ${tag_name} already exists."
+            index=$((index + 1))
+            tag_name="$(date +release-%Y-%m-%d).${index}"
+            echo "Trying tag ${tag_name}."
+          done
```
Comment (daniel-wong-dfinity-org) [inline .github/workflows/snapshot.yml:62]: If possible, comment the purpose/effect of this loop.

```suggestion
          # If there is already a tag with the same name, try appending ".${next_integer}".
          while git show-ref --tags --verify --quiet "refs/tags/${tag_name}"; do
            echo "Tag ${tag_name} already exists."
            index=$((index + 1))
            tag_name="$(date +release-%Y-%m-%d).${index}"
            echo "Trying tag ${tag_name}."
          done
```

PS: This seems orthogonal to the title. Maybe mention this too?

File: README.md [hunk 2]
```diff
@@ -37,14 +37,9 @@ the parameters passed to `ic-admin` from `bin/dfx-sns-sale-propose`.
...
 
-A new stock snapshot can be released by pushing a tag:
-
-```
-RELEASE="release-$(date +"%Y-%m-%d")"
-echo "$RELEASE"
-git tag "$RELEASE"
-git push origin "$RELEASE"
-```
+A new stock snapshot can be released by running the
```
Comment (daniel-wong-dfinity-org) [inline README.md:40]: No change.

I know the previous version said this, but I'm not sure what "stock snapshot" means.

Did the old instructions not work?

**Review by daniel-wong-dfinity-org — APPROVED** [review]

File: .github/workflows/snapshot.yml (hunk 1 above)
Comment (dskloetd) [reply .github/workflows/snapshot.yml:63]: Do you mean the PR title?
It's mentioned in the PR description, but it's not orhogonal.
If you only have scheduled releases on different dates, the tag names based on dates never collide.
But if you can create releases any time, you might create multiple on the same day, so we need this collision avoidance.

I've applied your comment suggestion. Thanks.

File: README.md (hunk 2 above)
Comment (dskloetd) [reply README.md:40]: "stock snapshot" is what we call the snsdemo snapshot in some places.
I guess the thought was that we'd have custom snapshots as well and have one "stock" snapshot which isn't custom.

The old instructions still work but the new instructions are simpler so it didn't seem useful to keep the old instructions.
